backend/src/src-bdpimport/src/main/java/com/bosch/osmi/sw360/bdp/datasink/thrift/ThriftUploader.java

```diff
@@ -0,0 +1,209 @@
+            return Collections.EMPTY_SET;
+        }
+
+        Set<String> releaseIds = componentsBdp.stream()
+                .map(c -> getOrCreateComponent(c, user))
+                .collect(Collectors.toSet());
+
+        if (releaseIds.size() != componentsBdp.size()) {
+            logger.warn("expected to get " + componentsBdp.size() + " different ids of releases but got " + releaseIds.size());
+        } else {
+            logger.info("The expected number of releases was imported.");
+        }
+
+        return releaseIds;
+    }
+
+    protected Optional<String> createProject(String bdpId, User user) throws TException  {
+        logger.info("Try to import bdp Project: " + bdpId);
+        logger.info("Sw360-User: " + user.email);
+        logger.info("Remote-Credentials: " + bdpApiAccessWrapper.getEmailAddress());
+
+        com.bosch.osmi.bdp.access.api.model.ProjectInfo projectBdp = bdpApiAccessWrapper.getProjectInfo(bdpId);
+        if (Objects.isNull(projectBdp)) {
+            logger.error("Unable to get Project from BDP Server named: " + bdpId);
+            return Optional.empty();
+        }
+
+        String bdpName = projectBdp.getProjectName();
+        if (getProjectId(bdpId, bdpName, user).isPresent()) {
+            logger.error("Project already in database: " + bdpId);
+            return Optional.empty();
+        }
+
+        Set<String> releaseIds = getOrCreateComponents(projectBdp, user);
+
+        Project projectSW360 = new BdpProjectInfoToSw360ProjectTranslator().apply(projectBdp);
+        projectSW360.setProjectResponsible(user.getEmail());
+        projectSW360.setReleaseIdToUsage(releaseIds.stream()
+                .collect(Collectors.toMap(Function.identity(), e -> "contained")));
+
+        return Optional.ofNullable(thriftExchange.addProject(projectSW360, user));
+    }
+
+    protected Optional<String> getProjectId(String bdpId, String bdpName, User user) throws TException  {
+
+        Project existingProject = thriftExchange.getAccessibleProjectByBdpId(bdpId, user);
+        if (!Objects.isNull(existingProject)) {
+            logger.info("Project to import was already imported with bdpId: " + bdpId);
+            return Optional.ofNullable(existingProject.getId());
+        }
+        existingProject = thriftExchange.getAccessibleProject(bdpName, user);
+        if (!Objects.isNull(existingProject)) {
+            logger.info("Project to import already exists in the DB with name: " + bdpName);
+            return Optional.ofNullable(existingProject.getId());
+        }
+        return Optional.empty();
+    }
+
+    public BdpImportStatus importBdpProjects(Collection<String> bdpProjectIds, User user) {
+        List<String> failedIds = new ArrayList<>();
+        List<String> successfulIds = new ArrayList<>();
+        BdpImportStatus bdpImportStatus = new BdpImportStatus().setRequestStatus(RequestStatus.SUCCESS);
+        Optional<String> projectId = Optional.empty();
```

Comment:
I'd move this declaration inside the loop. I doesn't present a problem now, but it is conceivable that if the code in the loop is changed, the value of the variable might get unintentionally reused between iterations.